Examples

meelod opened the pull request:
Added kmcos.create_kmc_model and kmcos_compile to all example files (now listed and named as ...__build.py. Tested and ran all __build.py files listed and created local_smart directories for those files.

kmcos committed:
Merge pull request #142 from kmcos/develop

Develop, create_kmc_model

meelod committed:
Examples

Added kmcos.create_kmc_model and kmcos_compile to all example files (now listed and named as ...__build.py

meelod commented:
@AdityaSavara Please take a look and let me know what files are not necessary for merging.

meelod commented:
MyThirdTPR__build.py and multispecies__build.py did not run completely from my testing.

[Edit: they do run, but have some waiting time, as indicated by the comments at the top of the files.]

meelod committed:
Examples

Removed local_smart directories that were not in previous kmcos example version

meelod commented:
Not Ready to Merge

meelod committed:
Comment corrections

meelod committed:
MyThirdTPR__build

AdityaSavara reviewed (changes requested):
There is a change that we should make here as well as in the intro2kmcos repository. We should keep this line everywhere:

kmc_model.backend = 'local_smart' #specifying is optional. 'local_smart' is the default. Currently, the other options are 'lat_int' and 'otf'

More importantly, there are some models where the build files have "otf" in the name. Those need the otf backend (not local_smart backend).

***

We should also keep this line in all our examples:
kmc_model.clear_model()  

***

Note that with the above two additions to each file, the correct order of the last commands during building are going to be like this:

kmc_model.backend = 'local_smart' #specifying is optional. 'local_smart' is the default. Currently, the other options are 'lat_int' and 'otf'
kmc_model.clear_model(model_name=kmc_model.model_name, backend=kmc_model.backend) #This line is optional: if you are updating a model, this line will remove the old model files (including compiled files) before exporting the new one. It is convenient to always include this line because then you don't need to 'confirm' removing/overwriting the old model during the compile step.
kmc_model.save_model()
kmcos.compile(kmc_model)

meelod committed:
kmc_model improvements

Added save_model() with a save() pointer for backwards compatibility. Added assignments to Project object in set meta rather than just the meta.

meelod committed:
Merge branch 'kmcos:master' into create_kmc_model

meelod commented:
> There is a change that we should make here as well as in the intro2kmcos repository. We should keep this line everywhere:
> 
> kmc_model.backend = 'local_smart' #specifying is optional. 'local_smart' is the default. Currently, the other options are 'lat_int' and 'otf'
> 
> More importantly, there are some models where the build files have "otf" in the name. Those need the otf backend (not local_smart backend).
> 
> We should also keep this line in all our examples: kmc_model.clear_model()
> 
> Note that with the above two additions to each file, the correct order of the last commands during building are going to be like this:
> 
> kmc_model.backend = 'local_smart' #specifying is optional. 'local_smart' is the default. Currently, the other options are 'lat_int' and 'otf' kmc_model.clear_model(model_name=kmc_model.model_name, backend=kmc_model.backend) #This line is optional: if you are updating a model, this line will remove the old model files (including compiled files) before exporting the new one. It is convenient to always include this line because then you don't need to 'confirm' removing/overwriting the old model during the compile step. kmc_model.save_model() kmcos.compile(kmc_model)

@AdityaSavara 
For kmc_model.clear_model, it provides an error whenever a local_smart folder does not already exist. I suggest we leave them in all files but comment them out in those where the local_smart for the __build.py files do not exist and inform to uncomment it after the local smart is created. Let me know your thoughts.

kmcos commented:
That means we should add a line of code to clear_model to check if the 
directory exists. From online :

 The simplest way to get a list of entries in a directory is to use 
os.listdir ().

kmcos commented:
So the syntax will be something like:

if desiredDirectory in os.listdir():
...
Else:
  pass

meelod committed:
Updates to kmcos examples

Reflects standard commands at the bottom of all __build.py files

meelod committed:
Merge branch 'create_kmc_model' of https://github.com/meelod/kmcos into create_kmc_model

meelod committed:
Updates to saving and clearing model

meelod committed:
Revert "Updates to kmcos examples"

This reverts commit eb2ed7607eab568db0e4e7d17fa09c8c263515ca.

meelod committed:
Updates to clear model

AdityaSavara reviewed (changes requested):
The additional xml files (really, all the xml files) should be removed.

meelod committed:
Removal of all xml files
